File: ispiti-vezbanje/sortiranje/selection-sort.c
#include <stdio.h>
#include <stdlib.h>

void selection_sort(int array[], int n){
    for (int i = 0; i < n; i++){
        int min = i;
        for (int j = i + 1; j < n ; j++){
            if (array[j] < array[min]){
                min = j;
            }
        }
        if (min != i){
            array[i] = array[i] ^ array[min];
            array[min] = array[i] ^ array[min];
            array[i] = array[i] ^ array[min];
        }
    }
}

int main(){
    int array[] = {6, 5, 4, 3, 2, 1, 7, 8, 0};
    int n = sizeof(array)/sizeof(array[0]);
    selection_sort(array, n);
    for (int i = 0; i < n; i++){
        printf("%d ", array[i]);
    }
    printf("\n");
    return 0;
}

File: ispiti-vezbanje/sortiranje/selection-sort.js
//create array of ranodm numbers
let arr = [];
for (let i = 0; i < 10; i++) {
    arr[i] = Math.floor(Math.random() * 20);
}
console.log(arr);


//create selection sort algorithm


//create selectionsort algorithm
// function selectionSort(arr) {
//     let min;
//     for (let i = 0; i < arr.length; i++) {
//         min = i;
//         for (let j = i + 1; j < arr.length; j++) {
//             if (arr[j] < arr[min]) {
//                 min = j;
//             }
//         }
//         if (min !== i) {
//             let temp = arr[i];
//             arr[i] = arr[min];
//             arr[min] = temp;
//         }
//     }
//     return arr;
// }

function selectionSort(arr) {
    let min
    for (let i = 0; i < arr.length; i++) {   
        min = i;
        for(let j = i + 1; j < arr.length; j++) {
            if(arr[j] < arr[min]){
                min = j;
            }
        }
        if(min !== i){
            arr[i] =  arr[i] ^ arr[min]; 
            arr[min] =  arr[i] ^ arr[min]; 
            arr[i] =  arr[i] ^ arr[min]; 
        }
    }
    return arr;
}

arr = selectionSort(arr);
console.log(arr);